File: app/src/main/java/com/bq/zowi/interactors/MeasureZowiBatteryLevelInteractorImpl.java
package com.bq.zowi.interactors;

import com.bq.zowi.controllers.ZowiDataController;
import java.util.concurrent.TimeUnit;
import rx.Single;
import rx.functions.Func1;

/* JADX INFO: loaded from: classes.dex */
public class MeasureZowiBatteryLevelInteractorImpl implements MeasureZowiBatteryLevelInteractor {
    private static final int RECEIVE_BATTERY_LEVEL_TIMEOUT_MILLIS = 500;
    private SendCommandToZowiInteractor sendCommandToZowiInteractor;
    private ZowiDataController zowiDataController;

    public MeasureZowiBatteryLevelInteractorImpl(SendCommandToZowiInteractor sendCommandToZowiInteractor, ZowiDataController zowiDataController) {
        this.sendCommandToZowiInteractor = sendCommandToZowiInteractor;
        this.zowiDataController = zowiDataController;
    }

    @Override // com.bq.zowi.interactors.MeasureZowiBatteryLevelInteractor
    public Single<Boolean> measureAndManageZowiBatteryLevel() {
        return this.zowiDataController.getBatteryLevel().flatMap(new Func1<Float, Single<? extends Boolean>>() { // from class: com.bq.zowi.interactors.MeasureZowiBatteryLevelInteractorImpl.1
            @Override // rx.functions.Func1
            public Single<? extends Boolean> call(Float receivedBatteryLevel) {
                return MeasureZowiBatteryLevelInteractorImpl.this.manageZowiBatteryLevel(receivedBatteryLevel.floatValue());
            }
        }).timeout(500L, TimeUnit.MILLISECONDS, Single.just(true));
    }

    @Override // com.bq.zowi.interactors.MeasureZowiBatteryLevelInteractor
    public Single<Boolean> manageZowiBatteryLevel(float batteryLevel) {
        return batteryLevel >= 50.0f ? Single.just(true) : Single.just(false);
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/usecases/MeasureZowiBatteryLevelInteractorImpl.kt
package com.bq.zowi.usecases

import com.bq.zowi.api.ZowiDataController
import io.reactivex.Single
import java.util.concurrent.TimeUnit

class MeasureZowiBatteryLevelInteractorImpl(
    private val zowiDataController: ZowiDataController
) : MeasureZowiBatteryLevelInteractor {

    override fun measureAndManageZowiBatteryLevel(): Single<Boolean> {
        return zowiDataController.getBatteryLevel()
            .flatMap { batteryLevel -> manageZowiBatteryLevel(batteryLevel) }
            .timeout(500L, TimeUnit.MILLISECONDS, Single.just(true))
    }

    override fun manageZowiBatteryLevel(batteryLevel: Float): Single<Boolean> {
        return Single.just(batteryLevel >= MeasureZowiBatteryLevelInteractor.BATTERY_LEVEL_LOWER_THRESHOLD)
    }
}
